### PR TITLE
Started Adding Documentation.

### PR DESCRIPTION
**Pull request by grokys**
This PR starts adding documentation for the library in the `docs` directory.

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Octokit.GraphQL.net
```
Comment (jcansdale) [inline docs/readme.md]: `.net` is pretty old-school. 😉 

image

Maybe use `.NET` or `.Net`?

File: docs/compiling-queries.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+In the [quickstart](quickstart.md) we created a query and ran it directly on the connection:
+
+```csharp
+var query = new Query().Viewer.Select(x => x.Login);
+var connection = new Connection("https://api.github.com/graphql", YOUR_OAUTH_TOKEN);
+var result = await connection.Run(query);
+```
+
+This is fine for queries you only want to run once, but each time this query is run, the query
+expression needs to be parsed and the GraphQL query created. If you're going to be re-using a
```
Comment (jcansdale) [inline docs/compiling-queries.md]: Is the bigger issue that parts of the expression will need to be compiled into non-GCable assemblies? This wouldn't be good if you're making lots of queries from a web server.

File: docs/paging.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    .Repository("octokit", "octokit.net")
+    .Issues()
+    .AllPages() // Auto-pages Issues
+    .Select(issue => new
+    {
+        issue.Number,
+        issue.Title,
+    }).Compile();
+
+var result = await Connection.Run(query);
```
Comment (jcansdale) [inline docs/paging.md]: I'm not 100% sure how auto-paging works under the hood. When I `foreach` my way through the result, will the next page be retrieved on demand or will all paged be retrieved before `Run` returns?

File: docs/compiling-queries.md (hunk 2 above)
Comment (grokys) [reply docs/compiling-queries.md]: Sorry, not sure what you mean here. Could you explain some more?

As far as I know, compiled expressions can be GC'd, if that was what you're worried about? https://stackoverflow.com/questions/5340201/expressiontdelegate-compile-and-garbage-collection

**Pull request merged**